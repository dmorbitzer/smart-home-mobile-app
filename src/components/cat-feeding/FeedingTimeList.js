import Icon from "@expo/vector-icons/MaterialCommunityIcons";
import { Text, Box, ListItem } from "@react-native-material/core";

import { formatStringToTime } from "../../api/DateTime";

export default function FeedingTimeList(props) {
  const sortedFeedingTimeData = props.feedingTimesData
    .slice()
    .sort((a, b) =>
      a.node.time > b.node.time ? 1 : b.node.time > a.node.time ? -1 : 0
    );
  const feedingList = sortedFeedingTimeData.map((element) => {
    if (
      element.node.cat.id === props.catId &&
      element.node.weekDay === props.weekDay
    ) {
      const time = formatStringToTime(element.node.time);
      return (
        <ListItem
          key={element.node.id}
          title={time}
          secondaryText={element.node.food.name}
          leading={
            <Icon name="silverware-fork-knife" color="black" size={25} />
          }
          onLongPress={() => {
            props.feedingTimeLongClick(element.node.id);
          }}
          delayLongPress={200}
        />
      );
    }
  });
  const isUndefined = (currentValue) => currentValue === undefined;
  if (feedingList.every(isUndefined)) {
    const message = props.weekDay
      ? "Keine Fütterungen an diesem Wochentag!"
      : "Bitte zuerst einen Wochentag wählen!";
    return (
      <Text style={{ alignSelf: "center", marginTop: 15 }}>{message}</Text>
    );
  }
  return <Box>{feedingList}</Box>;
}
