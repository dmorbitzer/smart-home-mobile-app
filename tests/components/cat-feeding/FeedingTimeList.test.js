import React from "react";
import renderer from "react-test-renderer";

import FeedingTimeList from "../../../src/components/cat-feeding/FeedingTimeList";
jest.useFakeTimers();

const props = {
  weekDay: 1,
  catId: "/api/cat/123",
  catName: "TEST",
  feedingTimesData: {
    data: {
      feedingTimes: {
        edges: [
          {
            node: {
              id: "/api/feeding_times/18",
              cat: {
                id: "/api/cats/246",
                name: "Chris",
              },
              food: {
                name: "DiÃ¤t Trockenfutter",
              },
              time: "1970-01-01T20:45:06+01:00",
              weekDay: 7,
            },
          },
          {
            node: {
              id: "/api/feeding_times/28",
              cat: {
                id: "/api/cats/246",
                name: "Chris",
              },
              food: {
                name: "Trockenfutter",
              },
              time: "1970-01-01T20:00:00+01:00",
              weekDay: 1,
            },
          },
        ],
      },
    },
  },
};
const feedingTimeLongClick = (feedingTimeId) => {
  console.log(feedingTimeId);
};
test("renders FeedingTime List correctly", () => {
  const tree = renderer
    .create(
      <FeedingTimeList
        feedingTimeLongClick={feedingTimeLongClick}
        weekDay={props.weekDay}
        feedingTimesData={props.feedingTimesData.data.feedingTimes.edges}
        catName={props.catName}
        catId={props.catId}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
