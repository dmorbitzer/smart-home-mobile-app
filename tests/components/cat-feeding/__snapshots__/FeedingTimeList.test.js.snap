// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FeedingTime List correctly 1`] = `
<Text
  style={
    [
      {
        "fontSize": 16,
        "letterSpacing": 0.5,
      },
      {
        "color": "#000000",
      },
      {
        "alignSelf": "center",
        "marginTop": 15,
      },
    ]
  }
>
  Keine FÃ¼tterungen an diesem Wochentag!
</Text>
`;
